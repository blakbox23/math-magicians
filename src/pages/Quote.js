import React from 'react';

function Quote() {
  return (
    <div className="quotesPg">
      <p data-testid="mathquote">
        Mathematics is not about numbers, equations, computations, or algorithms: it is about
        understanding.
        - William Paul Thurston
      </p>
    </div>
  );
}

export default Quote;
