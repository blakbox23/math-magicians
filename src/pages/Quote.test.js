import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('Quote', () => {
  test('test if it displays William Paul Thurston', () => {
    render(<Quote />);
    expect(screen.getByTestId('mathquote')).toHaveTextContent('Mathematics');
  });
});

test('Get a snapshot of Quote component', () => {
  const component = renderer
    .create(<Quote />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
