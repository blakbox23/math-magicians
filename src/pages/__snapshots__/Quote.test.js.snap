// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get a snapshot of Quote component 1`] = `
<div
  className="quotesPg"
>
  <p
    data-testid="mathquote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;
